src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -884,6 +904,31 @@ private static String findModuleName(Path modInfoPath) {
             }
         }
 
+        // returns the endianness of the target platform, if known. Else returns the
+        // current platform's endianness
+        private static ByteOrder getNativeEndianOfTargetPlatform(String targetPlatform) {
+            int index = targetPlatform.indexOf("-"); // of the form <operating system>-<arch>
+            if (index < 0) {
+                // unknown arch, return current platform's endianness
+                return ByteOrder.nativeOrder();
+            }
+            String archName = targetPlatform.substring(index + 1);
+            return switch (archName) {
+                case "x86", "x86_64",
+                        "alpha", "amd64",
+                        "arm", "aarch64",
+                        "ia64", "mipsel",
+                        "mips64el", "loongarch64",
+                        "ppc64le", "riscv32", "riscv64" -> ByteOrder.LITTLE_ENDIAN;
+                case "m68k", "mips",
+                        "mips64", "ppc",
+                        "ppc64", "s390",
+                        "s390x", "sh",
+                        "sparc", "sparcv9" -> ByteOrder.BIG_ENDIAN;
+                default -> ByteOrder.nativeOrder();
```

Comment:
When cross linking and the target platform is unknown then it might be better to have this fail rather than defaulting to the endianness of the current platform, otherwise this risks create a run-time image that is not usable.

I see you've listed a bunch of unknown platforms in the list. If getNativeEndianOfTargetPlatform fails when it encounters an unknown platform then help downstream porters to know that there is something to look at here.

Reply:
> When cross linking and the target platform is unknown then it might be better to have this fail rather than defaulting to the endianness of the current platform, otherwise this risks create a run-time image that is not usable.

I wasn't sure if we wanted to fail such cases when generating the image, but like you say, the generate image would be unusable anyway. I've now followed your suggestion to fail in such cases.

> I see you've listed a bunch of unknown platforms in the list. If getNativeEndianOfTargetPlatform fails when it encounters an unknown platform then help downstream porters to know that there is something to look at here.

I borrowed that list from our current build file here https://github.com/openjdk/jdk/blob/master/make/autoconf/platform.m4#L26. I think it's likely that it isn't the current supported list of platforms. I checked the `porters-dev` mailing list and I see that recently we have created this wiki which lists the supported platforms https://wiki.openjdk.org/display/HotSpot/Ports. Based on that I've trimmed down this list in the code (I continue to include `loongarch64` because I recently saw some PR which dealt with that architecture in mainline).  Having said that, it might be that this too isn't the fully accurate list, so please let me know if this needs additional changes.